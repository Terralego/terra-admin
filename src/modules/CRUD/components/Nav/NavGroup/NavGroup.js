import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Collapse, Button, Popover, PopoverInteractionKind, Position } from '@blueprintjs/core';

import NavItem from '../NavItem';
import NavIcon from '../NavIcon';

const NavGroup = ({
  navCollapsed,
  name: groupName,
  pictogram: groupPictogram,
  collapsed,
  views,
}) => {
  const [open, toggleGroup] = useState(collapsed);
  return (
    <li className="CRUD-nav__group">
      <Button
        className="CRUD-nav__group-button"
        alignText="left"
        onClick={() => toggleGroup(!open)}
        icon={<NavIcon isGroup src={groupPictogram} />}
        rightIcon={open ? 'chevron-up' : 'chevron-down'}
        minimal
        text={navCollapsed ? '' : groupName}
      />
      <Collapse
        isOpen={open}
      >
        <ul className="CRUD-nav__list">
          {views.length > 0 && views.map(view => (
            <li className="CRUD-nav__item" key={view.name}>
              {navCollapsed ? (
                <Popover
                  interactionKind={PopoverInteractionKind.HOVER}
                  position={Position.LEFT}
                  boundary="window"
                  content={<NavItem {...view} minified />}
                >
                  <NavIcon src={view.pictogram} />
                </Popover>
              ) : (
                <NavItem {...view} />
              )}
            </li>
          ))}
        </ul>
      </Collapse>
    </li>
  );
};

export default NavGroup;

NavGroup.propTypes = {
  navCollapsed: PropTypes.bool,
  name: PropTypes.string,
  pictogram: PropTypes.string,
  collapsed: PropTypes.bool,
  views: PropTypes.arrayOf(PropTypes.shape({})),
};

NavGroup.defaultProps = {
  navCollapsed: false,
  name: undefined,
  pictogram: undefined,
  collapsed: false,
  views: [],
};
