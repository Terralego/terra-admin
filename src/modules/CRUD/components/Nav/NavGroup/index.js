import NavGroup from './NavGroup';

export default NavGroup;
