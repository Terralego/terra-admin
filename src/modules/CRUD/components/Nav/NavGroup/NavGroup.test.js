import React from 'react';
import renderer from 'react-test-renderer';

import NavGroup from './NavGroup';

jest.mock('@blueprintjs/core', () => ({
  Collapse: ({ children, ...rest }) => <div {...rest}>{children}</div>,
  Button:  ({ children, ...rest }) => <div {...rest}>{children}</div>,
  Popover:  ({ children, ...rest }) => <div {...rest}>{children}</div>,
  PopoverInteractionKind: {
    HOVER: 'hover',
  },
  Position: {
    LEFT: 'left',
  },
}));

jest.mock('../NavItem', () => ({ children }) => <div>{children}</div>);
jest.mock('../NavIcon', () => props => <div {...props}>NavIcon</div>);

const views = [{
  id: 2,
  name: 'Foo',
  pictogram: null,
  order: 1,
  layer: {
    name: 'layerFoo',
  },
}, {
  id: 4,
  name: 'Bar',
  pictogram: 'path/to/image.png',
  order: 4,
  layer: {
    name: 'layerBar',
  },
}];

it('should snapshot correctly', () => {
  const tree = renderer.create((
    <NavGroup
      views={views}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should collapse correctly', () => {
  const tree = renderer.create((
    <NavGroup
      views={views}
      navCollapsed
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
