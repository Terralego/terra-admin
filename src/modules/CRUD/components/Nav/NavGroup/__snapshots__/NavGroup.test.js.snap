// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should collapse correctly 1`] = `
<li
  className="CRUD-nav__group"
>
  <div
    alignText="left"
    className="CRUD-nav__group-button"
    icon={
      <Unknown
        isGroup={true}
      />
    }
    minimal={true}
    onClick={[Function]}
    rightIcon="chevron-down"
    text=""
  />
  <div
    isOpen={false}
  >
    <ul
      className="CRUD-nav__list"
    >
      <li
        className="CRUD-nav__item"
      >
        <div
          boundary="window"
          content={
            <Unknown
              id={2}
              layer={
                Object {
                  "name": "layerFoo",
                }
              }
              minified={true}
              name="Foo"
              order={1}
              pictogram={null}
            />
          }
          interactionKind="hover"
          position="left"
        >
          <div
            src={null}
          >
            NavIcon
          </div>
        </div>
      </li>
      <li
        className="CRUD-nav__item"
      >
        <div
          boundary="window"
          content={
            <Unknown
              id={4}
              layer={
                Object {
                  "name": "layerBar",
                }
              }
              minified={true}
              name="Bar"
              order={4}
              pictogram="path/to/image.png"
            />
          }
          interactionKind="hover"
          position="left"
        >
          <div
            src="path/to/image.png"
          >
            NavIcon
          </div>
        </div>
      </li>
    </ul>
  </div>
</li>
`;

exports[`should snapshot correctly 1`] = `
<li
  className="CRUD-nav__group"
>
  <div
    alignText="left"
    className="CRUD-nav__group-button"
    icon={
      <Unknown
        isGroup={true}
      />
    }
    minimal={true}
    onClick={[Function]}
    rightIcon="chevron-down"
  />
  <div
    isOpen={false}
  >
    <ul
      className="CRUD-nav__list"
    >
      <li
        className="CRUD-nav__item"
      >
        <div />
      </li>
      <li
        className="CRUD-nav__item"
      >
        <div />
      </li>
    </ul>
  </div>
</li>
`;
